src/js/renderers/editor-dom.js

```diff
@@ -258,28 +307,31 @@ let destroyHooks = {
   },
 
   [IMAGE_SECTION_TYPE](renderNode, section) {
-    let post = renderNode.parent.postNode;
-    post.sections.remove(section);
-    renderNode.element.parentNode.removeChild(renderNode.element);
+    removeRenderNodeSectionFromParent(renderNode, section);
+    removeRenderNodeElementFromParent(renderNode);
   },
 
   [CARD_TYPE](renderNode, section) {
     if (renderNode.cardNode) {
       renderNode.cardNode.teardown();
     }
-    let post = renderNode.parent.postNode;
-    post.sections.remove(section);
-    renderNode.element.parentNode.removeChild(renderNode.element);
+    removeRenderNodeSectionFromParent(renderNode, section);
+    removeRenderNodeElementFromParent(renderNode);
   }
 };
 
 // removes children from parentNode that are scheduled for removal
 function removeChildren(parentNode) {
   let child = parentNode.childNodes.head;
+  let nextChild, method;
   while (child) {
-    let nextChild = child.next;
+    nextChild = child.next;
```

Comment:
Seems good, though also suspect is anything in the destroy hooks themselves is removing a renderNode. Render nodes should only be removed by the renderer itself (here on line 335, after a next child was read).